ecosystem/sep-0036.md

```diff
@@ -0,0 +1,137 @@
+
+## Stellar.toml
+
+Issuers will advertise the existence of an Approval Service through their stellar.toml file. This is done in the [[CURRENCIES]] section as different assets can have different requirements.
+
+### Fields:
+
+- `regulated` is a boolean indicating whether or not this is a regulated asset. If missing, `false` is assumed.
+- `approval_server` is the url of an approval service that signs validated transactions.
+- `approval_criteria` is a human readable string that explains the issuer's requirements for approving transactions.
+
+### Example
+
+```toml
+[[CURRENCIES]]
+code="GOAT"
+issuer="GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP"
+regulated=true
+approval_server="https://goat.io/tx_approve"
+approval_criteria="The goat approval server will ensure that transactions are compliant with NFO regulation"
+```
+
+## Approval Server
+
+The transaction approval server receives a signed transaction and checks for compliance and signs it on success.
+
+### Possible results
+- Success: Transaction was found compliant and signed by service.
+- Rejected: Transaction was rejected.
+
+### Request
+
+Transaction approval requests are executed by submitting an `HTTP POST` to `approval_server` with `Content-Type` being `application/x-www-form-urlencoded` or `application/json` and a single `transaction` parameter. `transaction` value is a base64 encoded transaction envelope XDR signed by the user. This is the transaction that will be tested for compliance and signed on success.
+
+### Responses
+
+HTTP responses have an `application/json` encoded body. All responses will contain, at least, a top level `status` parameter that indicates the type of the result.
+
+#### Success Response
+
+A Successful response will have a `200` HTTP status code and `success` as the `status` value. This response means that the transaction was found compliant and signed.
+
+Parameters:
+
+Name | Data Type | Description
+-----|-----------|------------
+status|string|`"success"`
+transaction|string|Transaction envelope XDR, base64 encoded. This transaction will have the original signature(s) from the request as well as the asset issuer's signature.
+message|string|A human readable string containing information to pass on to the user (optional).
+
+#### Rejected Response
+
+A Rejected response will have a `422` HTTP status code and `rejected` as the `status` value. It means that the transaction is not compliant and thus not signed by the server.
+
+The error message should explain why the transaction is not compliant and advice further actions for the client if possible. If there are further actions that the user must complete before this transaction can be approved, the issuer may provide a URL via `action_url` to faciliate the actions. If the issuer needs to asynchronously validate the transaction, the adviced wait time should be included in the error message if possible so the client would know when to retry.
```

Comment:
I'm not a fan of a single human readable error message in this situation, especially if important pieces of info like a wait time is going to be included.

I don't quite understand the `if the issuer needs to asynchronously validate the transaction` situation, can you explain what they would be validating and how it's async?

Finally, it sounds like the `action_url` is similar to the `interactive_url` returned by SEP-24 endpoints. If it is, I would explicitly mention that this URL must provide an intuitive UX for the user to navigate. In SEP-24, I think we require the `Content-Type` to be `text/html`.

Reply:
> I don't quite understand the if the issuer needs to asynchronously validate the transaction situation, can you explain what they would be validating and how it's async?

This is carried over from SEP-8. I don't know exactly how an issuer might do the validation to be honest.

Coupled with another comment
> I'm not a fan of a single human readable error message in this situation, especially if important pieces of info like a wait time is going to be included.

I could remove the mention about async validation and timeout or add a machine readable field `timeout` in the response. Which one do you prefer?

> In SEP-24, I think we require the Content-Type to be text/html.

Do you mean when an `action_url` is present, we want to use `text/html` instead of `application/json` as the content-type for the response? I can't find the requirement in SEP-24.